src/commands/executeFunction/executeEventGridFunction.ts

```diff
@@ -0,0 +1,156 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { type IActionContext, type IAzureQuickPickItem } from '@microsoft/vscode-azext-utils';
+import * as fs from 'fs-extra';
+import * as os from 'os';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { localize } from '../../localize';
+import { type FunctionTreeItemBase } from '../../tree/FunctionTreeItemBase';
+import { feedUtils } from '../../utils/feedUtils';
+import { type IFunction } from '../../workspace/LocalFunction';
+import { supportedEventGridSourceLabels, supportedEventGridSources, type EventGridSource } from './eventGridSources';
+import { executeFunctionWithInput } from './executeFunction';
+
+type FileMetadata = {
+    name: string;
+    path: string;
+    sha: string;
+    size: number;
+    url: string;
+    html_url: string;
+    git_url: string;
+    download_url: string;
+    type: string;
+    _links: {
+        self: string;
+        git: string;
+        html: string;
+    };
+};
+
+const sampleFilesUrl =
+    'https://api.github.com/repos/Azure/azure-rest-api-specs/contents/specification/eventgrid/data-plane/' +
```

Comment:
We should use an aka.ms link for this in case they ever change the repo or the repo formatting. That allows us to change this without having to do a new release.

I haven't tried it personally, but we should be able to just leave `{eventSource}` in the aka.ms link, and then just replace it like how you're already doing.

Reply:
Discussed offline. Tried using an `aka.ms` link, but turns out there isn't an easy way to do this parameterization with them. Keeping it like this for now.